app/Http/Controllers/FavoritesController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Models\Favorites;
+use Illuminate\Http\Request;
+use Illuminate\Session\Store as Session;
+
+class FavoritesController extends Controller
+{
+    public function store(Request $request, Session $session): void
+    {
+        $cityId = $request->input("city_id");
+
+        $existingFavorite = Favorites::where("user_id", $this->getUserId())
```

Comment:
You can get user instance from `$request` directly. 
```php
$request->user()->id;
```
`user()` might returns null if user is not logged in, so consider to use null safe operator `?` `user()?->id`
https://laravel.com/docs/10.x/authentication#retrieving-the-authenticated-user
https://stitcher.io/blog/php-8-nullsafe-operator

Reply:
Done